### ISSUE TITLE
[inductor] mix-order reduction: split column sum ignores xmask, wrong results when rows % XBLOCK != 0

### ISSUE THREAD

**Issue by eellison** [pytorch/pytorch#195845]
With `triton.mix_order_reduction` on (OSS default), a row and a column reduction over the same input fuse into one kernel. The column reduction is accumulated per RSPLIT chunk with `tl.sum(<value, 0)` without masking the tile by `xmask`. Padded rows load `other=0.0`, so any fused prologue that maps 0 to nonzero (`+ bias`, `exp`, softmax, ...) is summed once per padded row whenever `rows % XBLOCK != 0`.

```python
import torch

def f(x):
    y = x * 2 + 0.25
    return y.max(dim=-1).values, y.float().sum(dim=0)

x = torch.randn(40961, 129, device="cuda", dtype=torch.bfloat16)
_, colsum = torch.compile(f)(x)
ref = (x.float() * 2 + 0.25).sum(dim=0)
print((colsum - ref).abs().max().item())  # 0.75 (3 padded rows x 0.25); 0.0 for 40960 rows
```

cc @ezyang @gchanan @kadeng @msaroufim @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @muchulee8 @amjames @aakhundov @coconutruben @jataylo
